### PR TITLE
PB-7095 & PB-7096: Add libs for enabling cluster wide PSA

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:

- [x] Add func `runCmdOnceNonRoot()` to run commands without root
- [x] Add constants to hold file paths
- [x]  Add func `RunCmdsOnAllMasterNodes()` to run a command on all master nodes of a give cluster
- [x] Add func `ConfigureClusterLevelPSA()` to configure cluster level PSA setting on Vanilla cluster
- [x] Add func `RevertClusterLevelPSA()` to revert cluster level PSA configurations
- [x] Add func `VerifyClusterlevelPSA()` to verify if cluster level PSA settings are set on Kube-API server
- [x] Add dummy test `DummyPSATestcase` to test the libs 

**Which issue(s) this PR fixes** (optional)
Closes #PB-7095 PB-7096

Successful run link: 
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/163/console
~https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/157/console~
~https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/154/console~

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
...
+		fmt.Sprintf("mv %s %s", tempFilePath, kubeApiServerConfigFilePath),
+	}
+	log.Infof(fmt.Sprintf("%s", strings.Join(cmds, "\n")))
+	// Run the above set of commands in all the master nodes
+	err = RunCmdsOnAllMasterNodes(cmds)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(2 * time.Minute)
```
Comment (snigdha-px) [inline tests/common.go:13015]: Why are we adding sleep when we are using DoRetryWithTimeout?

File: tests/common.go [hunk 2]
```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
...
+	t := func() (interface{}, bool, error) {
+		if _, err := core.Instance().GetPods("kube-system", nil); err == nil {
+			return "", false, nil
+		}
+
+		return "", true, nil
+	}
+
+	_, err = task.DoRetryWithTimeout(t, kubeApiServerBringUpTimeout, 2*time.Minute)
+	if err != nil {
```
Comment (snigdha-px) [inline tests/common.go:13026]: Can we make 2*time.Minute configurable same as instead of hardcoding

File: tests/common.go [hunk 3]
```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
...
```
Comment (snigdha-px) [inline tests/common.go:13061]: same

**Review by snigdha-px — CHANGES_REQUESTED** [review]

File: tests/common.go [hunk 4]
```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
 	}
 	return multipathMap, nil
 }
+
+// RunCmdsOnAllMasterNodes Runs a set of commends on all the master nodes
```
[the remark this answers is not in the document]
Comment (kshithijiyer-px) [reply tests/common.go:12957]: Done

File: tests/common.go (hunk 2 above)
Comment (kshithijiyer-px) [reply tests/common.go:13029]: Done

File: tests/common.go (hunk 3 above)
Comment (kshithijiyer-px) [reply tests/common.go:13066]: Done

File: tests/common.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Done

File: tests/common.go [hunk 5]
```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
...
+	}
+
+	// Get the namespace where px-backup is present
+	pxBackupNamespace, err := backup.GetPxBackupNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Create a list of all the namespaces which need to be excluded
+	namespaces := []string{"default", "kube-system", pxBackupNamespace}
```
[the remark this answers is not in the document]
Comment (kshithijiyer-px) [reply tests/common.go:12989]: Done

**Review by snigdha-px — APPROVED** [review]
LGTM

**Comment by kshithijiyer-px** [comment]
Successful run after snigdha's comments: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/157/console

File: tests/common.go [hunk 6]
```diff
@@ -13111,3 +13131,133 @@ func CheckIopsandBandwidthinFA(flashArrays []pureutils.FlashArrayEntry, listofFa
 	}
 	return nil
 }
+
+// RunCmdsOnAllMasterNodes Runs a set of commands on all the master nodes
+func RunCmdsOnAllMasterNodes(cmds []string) error {
+	for _, node := range node.GetMasterNodes() {
+		for _, cmd := range cmds {
+			log.InfoD(fmt.Sprintf("Running command %s on %s", cmd, node))
```
Comment (ak-px) [inline tests/common.go:13139]: it should be node.Name

**Review by ak-px — APPROVED** [review]
LGTM, minor change.

File: tests/common.go (hunk 6 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Nice catch done.

File: tests/common.go [hunk 7]
```diff
@@ -13111,3 +13131,133 @@ func CheckIopsandBandwidthinFA(flashArrays []pureutils.FlashArrayEntry, listofFa
...
+	err = RunCmdsOnAllMasterNodes(cmds)
+	if err != nil {
+		return err
+	}
+
+	// Sleeping till the kubeAPI server comes up
+	time.Sleep(KubeApiServerWait)
+
+	// Wait for cluster to be in normal state
+	t := func() (interface{}, bool, error) {
```
Comment (mkoppal-px) [inline tests/common.go:13198]: How will this ensure the cluster is is normal state?

File: tests/common.go [hunk 8]
```diff
@@ -13111,3 +13131,133 @@ func CheckIopsandBandwidthinFA(flashArrays []pureutils.FlashArrayEntry, listofFa
...
+
+	_, err = task.DoRetryWithTimeout(t, kubeApiServerBringUpTimeout, KubeApiServerWait)
+	if err != nil {
+		return fmt.Errorf("API server didn't come up after change")
+	}
+	return nil
+}
+
+// VerifyClusterLevelPSA Verify if the cluster level PSA settings are set correctly
+func VerifyClusterlevelPSA() (bool, error) {
```
Comment (mkoppal-px) [inline tests/common.go:13251]: Can we just return error here?

**Review by mkoppal-px — APPROVED** [review]
LGTM except for a minor

File: tests/common.go (hunk 8 above)
Comment (kshithijiyer-px) [reply tests/common.go:13251]: Done

**Comment by kshithijiyer-px** [comment]
Latest run after Mithun's change https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/163/console

**Pull request merged**